server/src/main/java/org/elasticsearch/index/mapper/TermBasedFieldType.java

```diff
@@ -35,6 +37,10 @@ public TermBasedFieldType(
         super(name, isIndexed, isStored, hasDocValues, textSearchInfo, meta);
     }
 
+    protected boolean allowDocValueBasedQueries() {
```

Comment:
I was wondering about this new method: I initially thought this is an additional user knob to turn doc_value queries on and off, based on its naming (similar to allow expensive queries). I think I misunderstood and it only has to do with the fact that some field type that inherit from TermBasedFieldType don't support docvalue based queries, correct?

Reply:
Yeah, I changed my mind on this one while working on a follow-up. It disappears actually in the follow-up, see https://github.com/elastic/elasticsearch/pull/82925 (wait with review until full CI run). I was expecting some reuse in follow-ups, but that turned out more difficult, so I removed this extra abstraction

Reply:
++ sounds good thanks I will have a look when ready for review.